azavea/raster-vision-website2
# Add hero text and image carousel

**Overview**

This PR adds the basis of the hero component, including the headline text, image, carousel effect, and navigation buttons. The carousel image includes a fade effect and the transition is triggered either automatically after 5 seconds or via the navigation buttons.

**Notes**

This doesn't include the unique shape or other visuals of the hero component from the [designs](https://www.figma.com/file/dHNZK4IUu8W8QseGEkHBfY/RasterVision.IO-Website?node-id=0%3A1&t=i2Mh9Ep4tWjTMhpp-0). It also does not adjust for smaller screens yet, so you may see a bit of overlap between the text and images on narrow screens. These issues are the subject of #7 and #8, respectively. 

While I did search for possible libraries to use for the carousel effect, I found that many of the popular libraries weren't designed for our specific combination of text + image, where each has its own style of transition, are not pieces of a larger transitioning slide, and need to be triggered both automatically and via buttons but not via a swipe. As a result, this is built via jQuery and Javascript.

~The nuclei detection image has a watermark on it; I'll update that image when I have a replacement.~ I've replaced the nuclei detection image with a placeholder until a replacement is available.

**Testing Instructions**
- Run the server via `npm start` and go to localhost:8000 or view it at https://azavea.github.io/raster-vision-website2/
- Wait 5 seconds and make sure both the image and text change
- Use the buttons to navigate forwards and backwards and make sure they loop back through to the beginning or end as expected
- If you click a button, the counter resets and that the slide you click to still gets 5 seconds before transitioning
- The transition occurs without flickering or any other strange behavior and the general component looks generally like the designs

Resolves #6
